mods/cnc/maps/gdi05b/gdi05b.lua

```diff
@@ -0,0 +1,245 @@
+
+AllToHuntTrigger =
+{
+	Silo1, Proc1, Silo2, Silo3, Silo4, Afld1, Hand1, Nuke1, Nuke2, Nuke3, Fact1
+}
+
+GdiBaseDiscoveryTrigger =
+{
+	CPos.New(41,59), CPos.New(40,59), CPos.New(23,59), CPos.New(22,59),
+	CPos.New(41,58), CPos.New(40,58), CPos.New(23,58), CPos.New(22,58), 
+	CPos.New(41,57), CPos.New(40,57), CPos.New(23,57), CPos.New(22,57), 
+	CPos.New(41,56), CPos.New(40,56), CPos.New(23,56), CPos.New(22,56), 
+	CPos.New(41,55), CPos.New(40,55), CPos.New(23,55), CPos.New(22,55), 
+	CPos.New(41,54), CPos.New(40,54), CPos.New(23,54), CPos.New(22,54), 
+	CPos.New(41,53), CPos.New(40,53), CPos.New(23,53), CPos.New(22,53), 
+	CPos.New(41,52), CPos.New(40,52), CPos.New(23,52), CPos.New(22,52), 
+	CPos.New(41,51), CPos.New(40,51), CPos.New(23,51), CPos.New(22,51), 
+	CPos.New(41,50), CPos.New(40,50), CPos.New(41,49), CPos.New(40,49), 
+	CPos.New(39,59), CPos.New(38,59), CPos.New(25,59), CPos.New(24,59), 
+	CPos.New(39,58), CPos.New(38,58), CPos.New(25,58), CPos.New(24,58), 
+	CPos.New(39,57), CPos.New(38,57), CPos.New(25,57), CPos.New(24,57), 
+	CPos.New(39,56), CPos.New(38,56), CPos.New(25,56), CPos.New(24,56), 
+	CPos.New(39,55), CPos.New(38,55), CPos.New(25,55), CPos.New(24,55), 
+	CPos.New(39,54), CPos.New(38,54), CPos.New(25,54), CPos.New(24,54), 
+	CPos.New(39,53), CPos.New(38,53), CPos.New(25,53), CPos.New(24,53), 
+	CPos.New(39,52), CPos.New(38,52), CPos.New(25,52), CPos.New(24,52), 
+	CPos.New(39,51), CPos.New(38,51), CPos.New(25,51), CPos.New(24,51), 
+	CPos.New(39,50), CPos.New(38,50), CPos.New(39,49), CPos.New(38,49)
+}
+
+Atk1Delay = DateTime.Seconds(40)
+Atk2Delay = DateTime.Seconds(60)
+Atk3Delay = DateTime.Seconds(70)
+Atk4Delay = DateTime.Seconds(90)
+
+Atk1Waypoints = { waypoint4, waypoint5, waypoint6, waypoint7, waypoint8 }
+Atk2Waypoints = { waypoint0, waypoint1, waypoint2, waypoint3 }
+Atk3Waypoints = { waypoint4, waypoint5, waypoint6, waypoint7, waypoint8 }
+Atk4Waypoints = { waypoint0, waypoint1, waypoint2, waypoint3 }
+Atk5Waypoints = { waypoint0, waypoint1, waypoint2, waypoint3 }
+
+Atk5CellTriggers =
+{
+	CPos.New(17,55), CPos.New(16,55), CPos.New(15,55), CPos.New(50,54), CPos.New(49,54),
+	CPos.New(48,54), CPos.New(16,54), CPos.New(15,54), CPos.New(14,54), CPos.New(50,53),
+	CPos.New(49,53), CPos.New(48,53), CPos.New(50,52), CPos.New(49,52)
+}
+
+GdiBase = { GdiNuke1, GdiProc1, GdiWeap1, GdiNuke2, GdiPyle1, GdiSilo1, GdiSilo2, GdiHarv }
+GdiUnits = { "e2", "e2", "e2", "e2", "e1", "e1", "e1", "e1", "mtnk", "mtnk", "jeep", "jeep", "apc" }
+NodSams = { Sam1, Sam2, Sam3, Sam4 }
+
+getActors = function(owner, units)
```

Comment:
I was thinking out loud. If you will need exactly this function for all your other missions, then it may be best to implement this in C# and as you are providing a test case here, this pull request would be the best way to do it. History showed us that way complex helper tend to break on engine changes when they are not continually tested with CI. If this is only required for this special mission and not used anywhere else, then ignore this.